pyomo/common/tests/test_tee.py

```diff
@@ -51,10 +51,17 @@ def test_merge_out_and_err(self):
             # flush() and short pause should help
             t.STDOUT.write("Hello\nWorld")
             t.STDOUT.flush()
-            time.sleep(tee._poll_interval*20)
+            time.sleep(tee._poll_interval*10)
             t.STDERR.write("interrupting\ncow")
             t.STDERR.flush()
-            time.sleep(tee._poll_interval*30)
+            # For determinism, it is important that the STDERR message
+            # appears in hte output stream before we start shutting down
+            # the TeeStream (which will dump the OUT and ERR in an
+            # arbitrary order)
+            i = 0
+            while i < 1000 and 'cow' not in a.getvalue():
+                time.sleep(tee._poll_interval)
```

Comment:
Can we assert that `_poll_interval` has not been changed to a large value before starting this loop? I'm not sure why anyone would make it significantly larger, but it still concerns me a bit.

Reply:
Fixed.